opennebula/provider.go

```diff
@@ -113,9 +121,20 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		return nil, diags
 	}
 
-	oneClient := goca.NewDefaultClient(goca.NewConfig(username.(string),
+	tr := &http.Transport{}
+	insecure := d.Get("insecure")
+	if insecure.(bool) {
+		// goca allows for passing in a custom http.client,
+		// which allows us to conditionaly disable TLS verification for self-signed certs
+		tr = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+	}
```

Comment:
Maybe we can do something like this to avoid the `if`?

```go
insecure := d.Get("insecure")
tr = &http.Transport{
 	TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure.(bool)},
}
oneClient := goca.NewClient(goca.NewConfig(username.(string),
 		password.(string),
 		endpoint.(string)))
 		endpoint.(string)),
 		&http.Client{Transport: tr})
```